Civi/Financeextras/Hook/BuildForm/ContributionCreate.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Civi\Financeextras\Hook\BuildForm;
+
+use CRM_Financeextras_ExtensionUtil as E;
+use Civi\Financeextras\Utils\OptionValueUtils;
+
+class ContributionCreate {
+
+  /**
+   * @param \CRM_Contribute_Form_Contribution $form
+   */
+  public function __construct(private \CRM_Contribute_Form_Contribution $form) {
+  }
+
+  public function handle() {
+    $this->configureRecordPaymentField();
+    $this->preventUserFromSettingContributionStatus();
+  }
+
+  /**
+   * We are changing the CiviCRM core behavior that allow users
+   * to set contibution status manually during create/edit, instead
+   * this status will be set based on the payment allocated to the
+   * contribution.
+   */
+  private function preventUserFromSettingContributionStatus() {
+    try {
+      $statusElement = $this->form->getElement('contribution_status_id');
+      if (!$this->isEdit()) {
+        // By default new contribution will have a pending status
+        // and will be updated to the right status post payment(if any)
+        $statusElement->setValue(OptionValueUtils::getValueForOptionValue('contribution_status', 'Pending'));
+      }
+      $statusElement->freeze();
+    }
+    catch (\Throwable $th) {
+    }
+  }
+
+  /**
+   * CiviCRM by deafault always shows the payment fields for new contribution,
```

Comment:
typo in "deafault" , also can you add note that you also add extra fields like the "amount" ..etc which are not there by default.